### PR TITLE
feat: Edit NFT modal

### PR DESCRIPTION
**Pull request by BrickheadJohnny**
- Refactored the create NFT form, so we can re-use it both for the create and edit flow
- Refactored the `EditRewardAvailabilityModal` - moved the startTime/endTime inputs to a separate component so we can re-use them in the NFT form
- Refactored the `ImagePicker` component: we don't store the image as a `File` inside the form, we upload it instead and just save its URL to the form (this was necessary so we can easily re-use the same component inside the edit form)
- Implemented the edit NFT flow, it's quite tricky, because we need to save some data to the contract and to our API too (update `rolePlatform` & `guildPlatform` on 2 different endpoints) - let me know if you have a cleaner solution to it

### CODE REVIEW

File: src/platforms/ContractCall/hooks/useEditNft.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          editGuildPlatform.onSubmit({
+            platformGuildData: {
+              description: apiData.platformGuildData.richTextDescription,
+            },
+          })
+        )
+      }
+
+      // Handling the success/error state here, because we can't really "chain" these calls using our useEditGuildPlatform & useEditRolePlatform hooks
+      Promise.all(apiCalls)
```
Comment (BrickheadJohnny) [inline src/platforms/ContractCall/hooks/useEditNft.ts:76]: Please let me know if you have a better idea!

File: src/platforms/ContractCall/hooks/useEditNft.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+  const editRolePlatform = useEditRolePlatform({
+    rolePlatformId,
+  })
+
+  const editNftContractCalls = async (data: Partial<CreateNftFormType>) => {
+    const { contractData, apiData } = separateContractAndAPIData(data)
+
+    if (Object.keys(contractData).length > 0) {
+      // TODO: multicall
```
Comment (BrickheadJohnny) [inline src/platforms/ContractCall/hooks/useEditNft.ts:41]: Waiting for the new contract here.

File: src/platforms/ContractCall/ContractCallCardMenu.tsx [hunk 3]
```diff
@@ -17,22 +18,38 @@ const ContractCallCardMenu = ({ platformGuildId }: Props): JSX.Element => {
...
         <RemovePlatformMenuItem platformGuildId={platformGuildId} />
       </PlatformCardMenu>
 
-      <EditNFTDescriptionModal
-        isOpen={isOpen}
-        onClose={onClose}
-        guildPlatform={guildPlatform}
-      />
+      {isLegacy ? (
+        <EditNFTDescriptionModal
```
Comment (BrickheadJohnny) [inline src/platforms/ContractCall/ContractCallCardMenu.tsx:41]: We could technically re-use the new Edit NFT modal here, but that would require more internal logic in that component, so I added a ternary here instead. We'll probably drop support for older NFT rewards eventually, we'll be able to remove this logic then.

File: src/components/[guild]/RolePlatforms/components/EditRewardAvailabilityModal/components/StartEndTimeForm.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  if (!datetimeLocal) return null
+
+  try {
+    return new Date(datetimeLocal).toISOString()
+  } catch {
+    return null
+  }
+}
+
+const StartEndTimeForm = <TFieldValues, TContext>({
```
Comment (BrickheadJohnny) [inline src/components/[guild]/RolePlatforms/components/EditRewardAvailabilityModal/components/StartEndTimeForm.tsx:31]: Mostly copy-pasted it from `EditRewardAvailabilityModal`, just added generics to it, so we can use it in a typesafe way everywhere.

File: src/components/[guild]/RolePlatforms/components/EditRewardAvailabilityModal/EditRewardAvailabilityModal.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  isLoading?: boolean
+  onDone: (data: RolePlatformAvailabilityForm) => void
+}
+
+const AUTO_SUPPLY_PLATFORMS: Partial<Record<PlatformName, string>> = {
+  UNIQUE_TEXT: "secrets",
+  POAP: "minting links",
+}
+
+const EditRewardAvailabilityModal = ({
```
Comment (BrickheadJohnny) [inline src/components/[guild]/RolePlatforms/components/EditRewardAvailabilityModal/EditRewardAvailabilityModal.tsx:50]: Actually just renamed this file & extracted the startTime/endTime input from it.

**Review by dovalid — APPROVED** [review]

**Pull request merged**